### PR TITLE
A few enhancements to icalevents

### PR DESCRIPTION
**Pull request by ndw**
This PR contains the same changes as my earlier PR (#53) reworked so that they apply cleanly to the current master. Along the way, I made a few improvements so that all of the tests pass. In this PR:

* Fixed mangling of the timezone see https://github.com/irgangla/icalevents/issues/12#issuecomment-456121570
* Track all of the timezones defined; attempt to map events to the correct timezone
* Ignore dates listed as exceptions (i.e handle EXDATE on recurring events)
* Recompute the start time of recurring events in the current timezone (preventing recurring events from being off by an hour if the recurrence crosses the daylight savings time boundary)
* Make sure that attendee and organizer are always present

I've tried to document the reasoning behind each of the major changes.

### CODE REVIEW

**Comment by Hultner** [comment]
Very nice, looks like I can decommission my local patch when this is merged :)

File: icalevents/icalparser.py [hunk 1]
```diff
@@ -241,12 +250,24 @@ def parse_events(content, start=None, end=None, default_span=timedelta(days=7)):
...
+    for c in calendar.walk('VTIMEZONE'):
+        name = str(c['TZID'])
+        try:
+            timezones[name] = c.to_tz()
+        except IndexError:
+            # This happens if the VTIMEZONE doesn't
+            # contain start/end times for daylight
+            # saving time. Get the system pytz
+            # value from the name as a fallback.
+            timezones[name] = timezone(name)
```
Comment (Hultner) [inline icalevents/icalparser.py]: Would it be wise to to log a warning here?

File: icalevents/icalparser.py [hunk 2]
```diff
@@ -44,6 +45,8 @@ def __init__(self):
         self.created = None
         self.last_modified = None
         self.sequence = None
+        self.attendee = None
+        self.organizer = None
```
Comment (Hultner) [inline icalevents/icalparser.py]: Nice addition

File: icalevents/icalparser.py [hunk 3]
```diff
@@ -255,16 +276,110 @@ def parse_events(content, start=None, end=None, default_span=timedelta(days=7)):
...
+                        ecopy = e.copy_to(dtstart, e.uid)
+
+                        # We're effectively looping over the start time, we might need
+                        # to adjust the end time too, but don't have it's recurred value.
+                        # Make sure it's adjusted by constructing it from the meeting
+                        # duration. Pro: it'll be right. Con: if it was in a different
+                        # timezone from the start time, we'll have lost that.
+                        ecopy.end = dtstart + duration
+
+                    exdate = "%04d%02d%02d" % (ecopy.start.year, ecopy.start.month, ecopy.start.day)
```
Comment (Hultner) [inline icalevents/icalparser.py]: Hmm, not really a question for this PR but maybe more for the project as a whole but wouldn't it make sense to use the more and more readably `f"…"`-strings or `"…".format(…)`-strings if we want to support python 3.5 and prior versions? 

Python 3.5 is EOL later this year https://devguide.python.org/#status-of-python-branches 
If supporting it is still a priority we could use `.format()`, if not I'd go with f-strings.

File: icalevents/icalparser.py [hunk 4]
```diff
@@ -284,42 +399,40 @@ def parse_rrule(component, tz=UTC):
...
-            
+
             # Add exdates to the rruleset
             for exd in extract_exdates(component):
                 rule.exdate(exd)
-        
-        #TODO: What about rdates and exrules?
-        
+
+        # TODO: What about rdates and exrules?
```
Comment (Hultner) [inline icalevents/icalparser.py]: Is this something we should worry about?

File: icalevents/icalparser.py (hunk 1 above)
Comment (ndw) [reply icalevents/icalparser.py]: I don't think it's necessary. If the timezone isn't known locally, the call will fail. If it is known, the only case where it matters is if the rules for daylight saving time have changed.

File: icalevents/icalparser.py (hunk 3 above)
Comment (ndw) [reply icalevents/icalparser.py]: I don't feel strongly about it; but yes, doing it the same way throughout probably makes sense.

File: icalevents/icalparser.py (hunk 4 above)
Comment (ndw) [reply icalevents/icalparser.py]: Dunno. I don't think that's my comment, I must have just put the space in reflexively.

**Comment by irgangla** [comment]
@Hultner @ndw Do you want to maintain this project?

**Comment by ndw** [comment]
I'm happy to help, I'm not sure I have the bandwidth to take over.

**Comment by Hultner** [comment]
@irgangla I can certainly help maintaining it, if you want a new primary maintainer I have to think about that.

**Pull request merged**